api/src/main.rs

```diff
@@ -168,50 +165,71 @@ impl From<jwt_simple::Error> for Error {
     }
 }
 
-#[derive(Serialize, Deserialize, ToSchema, ToResponse)]
-#[aliases(
-    OkLoginResponse = OkResponse<LoginResponse>,
-    OkCreateListResponse = OkResponse<CreateListResponse>,
-    OkGetListsResponse = OkResponse<GetListsResponse>,
-    OkSearchAccountResponse = OkResponse<SearchAccountResponse>,
-    OkReadListResponse = OkResponse<ReadListResponse>,
-    OkAddToListResponse = OkResponse<AddToListResponse>,
-    OkGetHistoryResponse = OkResponse<GetHistoryResponse>,
-    OkUpdateItemResponse = OkResponse<UpdateItemResponse>,
-    OkDeleteItemResponse = OkResponse<DeleteItemResponse>,
-    OkDeleteListResponse = OkResponse<DeleteListResponse>,
-    OkUnshareResponse = OkResponse<UnshareResponse>,
-    OkGetSharesResponse = OkResponse<GetSharesResponse>,
-    OkShareListResponse = OkResponse<ShareListResponse>,
-    OkDeleteShareResponse = OkResponse<DeleteShareResponse>,
-    OkRecoveryInfoResponse = OkResponse<RecoveryInfoResponse>,
-    OkRecoverPasswordResponse = OkResponse<RecoverPasswordResponse>,
-    OkRegisterResponse = OkResponse<RegisterResponse>,
-    OkGetAccountNameResponse = OkResponse<GetAccountNameResponse>,
-    OkSetPublicResponse = OkResponse<SetPublicResponse>,
-    OkRemovePublicResponse = OkResponse<RemovePublicResponse>,
-    OkGetPantryResponse = OkResponse<GetPantryResponse>,
-    OkAddToPantryResponse = OkResponse<AddToPantryResponse>,
-    OkRefillPantryResponse = OkResponse<RefillPantryResponse>,
-    OkEditPantryItemResponse = OkResponse<EditPantryItemResponse>,
-    OkDeletePantryItemResponse = OkResponse<DeletePantryItemResponse>,
-)]
-struct OkResponse<T> {
-    ok: T,
+trait OkResponse {
+    type Wrapper;
+
+    fn ok(ok: Self) -> Rsp<Self>;
 }
 
-impl<T> OkResponse<T> {
-    fn ok(v: T) -> Rsp<T> {
-        Ok(Json(Self { ok: v }))
-    }
+macro_rules! alias {
+    ($($okResp:ident => $ty:ident),* $(,)?) => {
+        pub mod ok_response {
+            use kabalist_types::*;
+            $(
+
+            #[derive(crate::Serialize, crate::Deserialize, crate::ToSchema, crate::ToResponse)]
+            pub struct $okResp {
+                ok: $ty,
```

Comment:
If we need to have $ty availaible we could do `kabalist_types::$ty`

Reply:
This would make the macro not as versatile. but seeing the use case here it should be fine.